### PR TITLE
Snapshot restore - various UI updates, input validation debugging

### LINKED ISSUE #243: [Restore Snapshots/Indexes]
**Is your feature request related to a problem?**
In the 2.1 release, we introduced Snapshots Management with the ability to automatically create snapshots to backup user data from OpenSearch indexes. Today, users have to restore snapshots through the restore API. As a follow-up, we want to introduce restore functionality in Dashboards UI. Users looking for a guided experience will be able to restore their indexes from existing snapshots within OpenSearch Dashboards.

**What solution would you like?**
Add restore option on the Snapshots page on UI to allow users to select a snapshot to restore. 
Support restore parameters already available in the Restore API on UI. Specifically, we would like to allow restoring indices to a new index name. 
Support Restore Activity tab to monitor ongoing restore.

### PR DESCRIPTION
**Pull request by phestecAMZN**
### Description
This PR addresses requests made yesterday by UX as well as debugging of user input validation.
see images.
for validation - 

- addressed incorrect ui when switching on and of certain options

### Issues Resolved
#243

### Check List
- [ x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
Screen Shot 2022-11-03 at 12 36 08 PM
Screen Shot 2022-11-03 at 12 38 23 PM
Screen Shot 2022-11-02 at 4 57 19 PM

### CODE REVIEW

File: public/pages/Snapshots/components/RestoreSnapshotFlyout/RestoreSnapshotFlyout.tsx [hunk 1]
```diff
@@ -331,18 +280,48 @@ export default class RestoreSnapshotFlyout extends Component<RestoreSnapshotProp
   };
 
   onToggle = (e: ChangeEvent<HTMLInputElement>) => {
-    const { restore_specific_indices, restore_all_indices } = RESTORE_OPTIONS;
+    const { restore_specific_indices, restore_all_indices, customize_index_settings, ignore_index_settings } = RESTORE_OPTIONS;
 
     if (e.target.id === restore_specific_indices) {
+      console.log("specific");
```
Comment (bowenlan-amzn) [inline public/pages/Snapshots/components/RestoreSnapshotFlyout/RestoreSnapshotFlyout.tsx:286]: remove

File: public/pages/Snapshots/components/RestoreSnapshotFlyout/RestoreSnapshotFlyout.tsx [hunk 2]
```diff
@@ -141,21 +142,36 @@ export default class RestoreSnapshotFlyout extends Component<RestoreSnapshotProp
...
+    if (options.ignore_index_settings?.length === 0) {
+      delete options.ignore_index_settings;
     }
 
-    if (options.index_settings.length === 0) {
-      delete options.index_settings;
-    } else {
-      const badJSON = this.checkBadJSON(customIndexSettings);
-      this.setState({ badJSON })
+    console.log(options);
```
Comment (bowenlan-amzn) [inline public/pages/Snapshots/components/RestoreSnapshotFlyout/RestoreSnapshotFlyout.tsx:153]: remove

File: public/pages/Snapshots/components/RestoreSnapshotFlyout/RestoreSnapshotFlyout.tsx [hunk 3]
```diff
@@ -141,21 +142,36 @@ export default class RestoreSnapshotFlyout extends Component<RestoreSnapshotProp
...
-    const badIgnore = this.checkCustomIgnoreConflict();
-    const noIndicesSelected = this.checkNoSelectedIndices(options.indices);
+    if (options.index_settings?.length === 0) {
+      delete options.index_settings;
+    }
 
-    if (badPattern || badRename || badIgnore || noIndicesSelected) {
-      this.setState({ badPattern, badRename, badIgnore, noIndicesSelected });
-      return;
+    if (options.ignore_index_settings?.length === 0) {
```
Comment (bowenlan-amzn) [inline public/pages/Snapshots/components/RestoreSnapshotFlyout/RestoreSnapshotFlyout.tsx:149]: Should we use == here?

File: public/pages/Snapshots/components/RestoreSnapshotFlyout/RestoreSnapshotFlyout.tsx (hunk 1 above)
Comment (phestecAMZN) [reply public/pages/Snapshots/components/RestoreSnapshotFlyout/RestoreSnapshotFlyout.tsx]: Done

File: public/pages/Snapshots/components/RestoreSnapshotFlyout/RestoreSnapshotFlyout.tsx (hunk 3 above)
Comment (phestecAMZN) [reply public/pages/Snapshots/components/RestoreSnapshotFlyout/RestoreSnapshotFlyout.tsx]: sure. Done

File: public/pages/Snapshots/components/RestoreSnapshotFlyout/RestoreSnapshotFlyout.tsx (hunk 2 above)
Comment (phestecAMZN) [reply public/pages/Snapshots/components/RestoreSnapshotFlyout/RestoreSnapshotFlyout.tsx]: Done

**Review by bowenlan-amzn — APPROVED** [review]

**Review by AWSHurneyt — APPROVED** [review]

**Pull request merged**